spec.bs

```diff
@@ -221,6 +226,60 @@ The <dfn for="interest group">estimated size</dfn> of an [=interest group=] |ig|
     [=interest group ad/render url=].
   1. The [=string/length=] of |ad|'s [=interest group ad/metadata=] if the field is not null.
 
+<div algorithm>
+
+To <dfn>check interest group permissions</dfn> given an [=origin=]
+|ownerOrigin|, an [=origin=] |frameOrigin|, and a [=boolean=] |isJoin|:
+1. If |ownerOrigin| is [=same origin=] to |frameOrigin|, then return true.
+1. Let |permissionsUrl| be the result of [=building an interest group permissions url=]
+  with |ownerOrigin| and |frameOrigin|.
+1. Let |request| be the result of [=creating a request=] with |permissionsUrl|, [=create a request/accept=]
+  set to "`application/json`"|. TODO: [maybe](https://source.chromium.org/chromium/chromium/src/+/main:content/browser/interest_group/interest_group_permissions_checker.cc;drc=adac219925ef5d9c9a954d189c2e4b8852a4bbed;l=123) require cors mode here?
+1. Let |resource| be null.
+1. [=Fetch=] |request| with [=fetch/processResponseConsumeBody=] set to the following steps given a
+  [=response=] |response| and |responseBody|:
+  1. If |responseBody| is null or failure, set |resource| to failure and return.
+  1. Let |headers| be |response|'s [=response/header list=].
+  1. If [=header list/getting a structured field value=] "X-Allow-FLEDGE" from |headers| does not
+    return true, set |resource| to failure and return.
+  1. Let |headerMimeType| be the result of [=header list/extracting a MIME type=] from |headers|.
+  1. Set |resource| to failure and return, if any of:
+    1. |headerMimeType| is failure.
+    1. |headerMimeType| is not a [=JSON MIME type=].
+  1. Let |mimeTypeCharset| be |headerMimeType|'s [=MIME type/parameters=]["`charset`"].

```

Comment:
Didn't find code that checks charset for the response. [code](https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:content/browser/interest_group/interest_group_permissions_checker.cc;drc=3b6a047da8255b32ec49f7c5476ed8429118b358;l=171)

The spec's [[=fetch resource=]](https://github.com/WICG/turtledove/blob/main/spec.bs#L844) does check charset ([code](https://crsrc.org/c/content/services/auction_worklet/auction_downloader.cc;drc=0e9a0b6e9bb6ec59521977eec805f5d0bca833e0;l=248))

Reply:
I copied this from the `[=fetch resource=]` algorithm. Why does it make sense to check charset in one case and not the other?

Reply:
I don't have a good answer to why there's such a difference. I asked in the chat room, maybe we can follow up there. But from my understanding of the code, it seems we're not checking charset for .well-known, so probably remove it here, and we can make changes to it later if needed.

Reply:
I followed your advice in the chat to copy from https://github.com/fedidcg/FedCM/blob/main/spec/index.bs